### PR TITLE
[WIP] Adapt node create/serve commands for Windows

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

This PR covers the Node commands for creating / serving projects natively on Windows.

### WHAT is this pull request doing?

Replace `fork` with `spawn` when running supervised processes (such as ngrok for serving), since `fork` is not natively supported on Windows.

### CODE REVIEW

File: lib/shopify-cli/process_supervision.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class ProcessSupervision
     # a string or a symbol to identify this process by
     attr_reader :identifier
     # process ID for the running process
-    attr_reader :pid
+    attr_accessor :pid
```
Comment (andyw8) [inline lib/shopify-cli/process_supervision.rb:14]: Why the change to this?

Comment (andyw8) [reply lib/shopify-cli/process_supervision.rb:14]: Ignore, I see it's being set from outside.

File: lib/shopify-cli/process_supervision.rb [hunk 2]
```diff
@@ -104,10 +110,12 @@ def running?(identifier)
       end
     end
 
-    def initialize(identifier, pid:, time: Time.now.strftime('%s')) # :nodoc:
+    def initialize(identifier, pid: nil, time: Time.now.strftime('%s')) # :nodoc:
```
Comment (tanema) [inline lib/shopify-cli/process_supervision.rb:113]: why can pid be nil now?

Comment (paulomarg) [reply lib/shopify-cli/process_supervision.rb:113]: The pid can be nil because we need this object before we call `spawn`, which returns the pid that gets set to it (via the new accessor above).
